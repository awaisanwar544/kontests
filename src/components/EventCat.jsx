import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { useSelector } from 'react-redux';

const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-content: center;
  align-items: center;
  padding: 2em;
  background-color: #fd5294;
  text-align: center;

  h2 {
    font-size: 3em !important;
    font-weight: 900;
    color: #ffffffd0;
  }

  span {
    color: #c31d5d;
  }
`;

const EventCat = ({ category }) => {
  const events = useSelector((state) => state.events);
  return (
    <Wrapper>
      <h2>
        {category}
        <br />
        <span>{events[category].length}</span>
      </h2>
    </Wrapper>
  );
};

EventCat.propTypes = {
  category: PropTypes.string.isRequired,
};

export default EventCat;
