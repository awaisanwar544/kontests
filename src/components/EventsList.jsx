import React from 'react';
import PropTypes from 'prop-types';
import { v4 as uuidv4 } from 'uuid';
import { useSelector } from 'react-redux';
import styled from 'styled-components';

const Wrapper = styled.div`
  background-color: #fd5294;
`;

const EventItem = styled.div`
  background-color: #ffffff83;
  padding: 1em;
  margin: 1em 0;
  font-size: 1em;

  div {
    display: flex;
    justify-content: space-between;
  }

  span {
    font-weight: 900;
    text-decoration: underline;
  }

  ul {
    list-style: none;
  }

  li {
    margin: 1em 0;
  }
`;

const EventsList = ({ site }) => {
  const events = useSelector((state) => state.events[site]);
  return (
    <Wrapper>
      {events.map((item) => (
        <EventItem key={uuidv4()}>
          <div>
            <h2>{item.name}</h2>
            {site === 'All Events' && (<h3>{item.site}</h3>)}
            <h3>{(item.status === 'CODING') ? 'Ongoing' : 'Not Started'}</h3>
          </div>
          <ul>
            <li>
              <span>Starting From:</span>
              {' '}
              {Date(Date.parse(item.start_time))}
            </li>
            <li>
              <span>Ends:</span>
              {' '}
              {Date(Date.parse(item.end_time))}
            </li>
            <li>
              <a target="_blank" rel="noreferrer" href={item.url}>Official Link</a>
            </li>
          </ul>
        </EventItem>
      ))}
    </Wrapper>
  );
};

EventsList.propTypes = {
  site: PropTypes.string.isRequired,
};

export default EventsList;
