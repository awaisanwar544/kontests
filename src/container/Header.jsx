import React from 'react';
import styled from 'styled-components';
import { NavLink } from 'react-router-dom';

const Wrapper = styled.header`
  display: flex;
  justify-content: space-between;
  align-items: center;

  h1 {
    font-size: 3em;
    color: #ffffffd0;
  }

  a {
    font-size: 3em;
    color: #ffffffd0;
    text-decoration: none;
  }
`;

const Header = () => (
  <Wrapper>
    <div>
      {window.location.pathname === '/' && (<NavLink to="/">&#60; Home</NavLink>)}
    </div>
    <div>
      <h1>Online Tech Events</h1>
    </div>
    <div />
  </Wrapper>
);

export default Header;
