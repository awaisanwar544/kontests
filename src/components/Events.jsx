import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';
import { getEvents } from '../redux/eventsReducer';
import EventCat from './EventCat';

const MainWrapper = styled.div`
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;

  h2 {
    color: #ffffff;
    font-size: 2em;
  }
  a {
    text-decoration: none;
  }
`;

const ListWrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  column-gap: 0.5em;
  row-gap: 0.5em;
`;

const Events = () => {
  const events = useSelector((state) => state.events);
  const dispatch = useDispatch();
  const list = Object.keys(events).filter((el) => el !== 'All Events');
  useEffect(() => {
    if (list.length === 0) {
      dispatch(getEvents());
    }
  }, []);
  return (
    <MainWrapper className="App">
      {events['All Events'] && (
        <NavLink to="AllEvents"><EventCat category="All Events" /></NavLink>
      )}
      <h2>List by Sites</h2>
      <ListWrapper>
        {list.length > 0 && (
          list.map((item) => <NavLink key={item} to={item.replace(/\s/g, '')}><EventCat category={item} /></NavLink>)
        )}
      </ListWrapper>
    </MainWrapper>
  );
};

export default Events;
