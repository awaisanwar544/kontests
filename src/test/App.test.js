import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import Events from '../components/Events';
import Header from '../components/Header';
import EventCat from '../components/EventCat';
import EventsList from '../components/EventsList';

jest.mock('react-redux');

describe('App Components Render Correctly', () => {
  it('Header renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Events renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Events />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('EventCat renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <EventCat category="" />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('EventsList renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <EventsList site="" />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
