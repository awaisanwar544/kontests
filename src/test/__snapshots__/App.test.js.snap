// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components Render Correctly EventCat renders correctly 1`] = `null`;

exports[`App Components Render Correctly Events renders correctly 1`] = `null`;

exports[`App Components Render Correctly EventsList renders correctly 1`] = `null`;

exports[`App Components Render Correctly Header renders correctly 1`] = `
<header
  className="sc-hKMtZM jFqUXH"
>
  <div>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
    >
      &lt; Home
    </a>
  </div>
  <div>
    <h1>
      Online Tech Events
    </h1>
  </div>
  <div />
</header>
`;
